### PR TITLE
Implement the inbox peek opcode

### PR DESCRIPTION
**Pull request by hkalodner**
- Adds a pending inbox field to the VM state
- Adds the `inboxpeek` opcode at 0x71

### CODE REVIEW

**Comment by hkalodner** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones removed
==============
+ packages/arb-avm-cpp/tests/checkpoint.cpp  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/OffchainLabs/arbitrum/pullRequest?prid=5970771&bid=19578148)

File: docs/AVM_Specification.md [hunk 1]
```diff
@@ -229,7 +232,8 @@ The instructions are as follows:
...
-| 0x72                                            | inbox         | Block until the VM's inbox sequence is non-empty. Then remove the first item from the inbox sequence as supplied by the Runtime Environment and push the result onto the Data Stack.                                                                                                                                                                                                                                                                                                     | 40          |
+| 0x71                                            | inboxpeak     | Pop a Value (A) off the Data Stack. If A is not an integer, raise an Error. Otherwise if the Pending Message is the empty Tuple then (block until the VM's inbox sequence is non-empty, then remove the first item from the inbox sequence as supplied by the Runtime Environment and set the Pending Message equal to it). Then if A equals the second value in the Pending Message Tuple, Push 1 onto the Data Stack, otherwise push 0 onto the Data Stack.                            | 40          |
```
Comment (edfelten) [inline docs/AVM_Specification.md:235]: It should be "peek", not "peak"

**Comment by hkalodner** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones removed
==============
+ packages/arb-avm-cpp/tests/checkpoint.cpp  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/OffchainLabs/arbitrum/pullRequest?prid=5970771&bid=19578148)

File: docs/AVM_Specification.md (hunk 1 above)
Comment (hkalodner) [reply docs/AVM_Specification.md]: Fixed

**Comment by hkalodner** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones removed
==============
+ packages/arb-avm-cpp/tests/checkpoint.cpp  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/OffchainLabs/arbitrum/pullRequest?prid=5970771&bid=19578148)

**Pull request merged**